matchingengine/OrderBook_test.go

```diff
@@ -53,3 +53,197 @@ func TestLoadOldAsk(t *testing.T) {
 	ob.LoadOldAsk(limitAsk)
 
 }
+
+func TestLoadOldAskStopLoss(t *testing.T) {
+
+	// This initializes the logger. This is required because LoadOldAsk tries accessing the logger
+	// and if it is not initialized, a nil reference error is thrown.
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl := gomock.NewController(t)
+	defer mockControl.Finish()
+
+	var stockID uint32 = 1
+	var stockQuantity uint32 = 10
+	var stockPrice uint32 = 20
+
+	mockAskQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidQueue := mocks.NewMockBidPQueue(mockControl)
+	mockAskStoplossQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidStoplossQueue := mocks.NewMockBidPQueue(mockControl)
+	mockDepth := mocks.NewMockMarketDepthStream(mockControl)
+
+	ob := &orderBook{
+		logger: utils.Logger.WithFields(logrus.Fields{
+			"module":        "matchingengine.OrderBook.test",
+			"param_stockId": stockID,
+		}),
+		stockId:     stockID,
+		askChan:     make(chan *models.Ask),
+		bidChan:     make(chan *models.Bid),
+		asks:        mockAskQueue,
+		bids:        mockBidQueue,
+		askStoploss: mockAskStoplossQueue,
+		bidStoploss: mockBidStoplossQueue,
+		depth:       mockDepth,
+	}
+
+	stopLossAsk := makeAsk(1, stockID, models.StopLoss, stockQuantity, stockPrice, "")
+
+	mockAskStoplossQueue.EXPECT().Push(stopLossAsk).Times(1)
+
```

Comment:
Does removing the check for `AddOrder` ensure that the test fails if it is called? Or do we need to add `Times(0)` in order to ensure it?